.github/workflows/pipeline.yml

```
@@ -0,0 +1,98 @@
+        run: semgrep --config p/default --config p/typescript --config p/javascript --error --metrics=off
+
+      # Qwen step intentionally omitted — runs locally only via
+      # scripts/pre-pr.sh (spec step 15). CI relies on CodeRabbit + CodeAnt
+      # for the post-push AI review perspective.
+
+  sonar:
+    name: SonarQube
+    runs-on: ubuntu-latest
+    # Skip cleanly when the secret isn't configured — self-hosted Sonar
+    # isn't reachable from GHA unless tunneled (we use Cloudflare Tunnel).
+    if: ${{ secrets.SONAR_TOKEN != '' }}
```

actionlint (1.7.12)

[error] context "secrets" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)